### DISCUSSION TITLE
Wrapping clap for other formats

### DISCUSSION THREAD

**Discussion by SamWindell** [free-audio/clap#190] [General]
First of all, I'm delighted to see this new format! A simple, well designed C API is exactly what is needed for a plugin format IMO.

I'm evaluating whether to change my plugin to use CLAP as its main API, and then creating wrapper layers onto of clap for the other formats that I have to support (VST3, AU, etc.). I need to transition my plugin from an old style VST2 infrastructure to more modern APIs.

The webpage says that wrappers is something that some people have already started and could be open source too. Is there any news on that? And do you have any advice for what I'm trying to achieve?

**Comment by urshimself** [comment]
Hi Sam,

Thank you. That sounds great, and it's exactly what we had in mind!

The first wrapper (CLAP-as-VST3) is ready for testing on macOS and Windows, with a few minor tasks left on Linux:

https://github.com/defiantnerd/clap-wrapper

When compiled, it is a 300k binary. You need to name it to the CLAP plug-in you wish to wrap, and then it'll load the CLAP and expose itself as a VST3 version of said plug-in.

Further extensions for AU are next on the list.

**Reply by SamWindell** [reply]
Amazing! Thanks for the info Urs.

This is very promising indeed. I'll put the CLAP-as-VST3 wrapper through it's paces as I work on this over the next few weeks.

**Comment by JBetz** [comment]
Has anyone tried going the other direction? I.e., VST plugins wrapped with a CLAP interface.

**Reply by baconpaul** [reply]
I started making a clap from aus idly a little while ago when I was thinking of writing a clap inky host but didn’t finish it. Don’t know if others have tried. We have a start of clap hosting in juce also which would sort of let you do that with some juce code but it is early.

**Comment by defiantnerd** [comment]
clap-wrapper has now reached a mature level in the mean time.
